tests/add-to-cart.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { test, expect } from '@playwright/test';
+
+
+test('test', async ({ page }) => {
+  await page.goto('https://practicesoftwaretesting.com/');
+  await page.locator('[data-test="product-01JR2SQFT1FHCQQD35GS5JAMHC"]').click();
+  expect(page.url()).toContain('https://practicesoftwaretesting.com/product');
```

Comment:
I'd use the assertion with retries https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url